Wrappers/Python/cil/optimisation/functions/L1Norm.py

```diff
@@ -187,5 +186,117 @@ def proximal(self, x, tau, out = None):
 
         return soft_shrinkage(x, tau, out = out) 
 
+class WeightedL1Norm(Function):
+    r"""WeightedL1Norm function
+            
+            .. math:: F(x) = ||x||_{\ell^1(w)} (w is array of positive weights)
+    
+    Parameters:
+    -----------
+
+    weight: weight array matching the size of the wavelet coefficients, default None
+        If None returns the regular L1Norm.
+    b: data, default None
+        Translation of the function.
+    """
+
+    def __new__(cls, weight=None, b=None):
+            
+        if weight is None:
+            return L1Norm(b=b)
+        else:
+            return super(WeightedL1Norm, cls).__new__(_WeightedL1Norm)
+
+class _WeightedL1Norm(WeightedL1Norm): 
+           
+    def __init__(self, weight=None, b=None):
+        super(WeightedL1Norm, self).__init__()
+        self.weight = weight
+        self.b = b
+
+        if np.min(weight) <= 0:
+            raise ValueError("Weights should be strictly positive!")
```

Comment:
I this true? Can't we have 0s and use it as a mask?

Reply:
we could but the code is currently using the weights at the denominator so 0s are bad.